tests/unit/scripts/test_lint.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python3
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Tests `cylc lint` CLI Utility.
+"""
+import difflib
+from pathlib import Path
+import pytest
+import re
+
+from cylc.flow.scripts.lint import (
+    CHECKS,
+    check_cylc_file,
+    get_cylc_files,
+    get_reference,
+    parse_checks
+)
+
+
+TEST_FILE = """
+[visualization]
+
+[cylc]
+    include at start-up = foo
+    exclude at start-up = bar
+    log resolved dependencies = True
+    required run mode = False
+    health check interval = PT10M
+    abort if any task fails = true
+    suite definition directory = '/woo'
+    disable automatic shutdown = false
+    reference test = true
+    spawn to max active cycle points = false
+    [[simulation]]
+        disable suite event handlers = true
+    [[authentication]]
+    [[environment]]
+    force run mode = dummy
+    [[events]]
+        abort on stalled = True
+        abort if startup handler fails= True  # deliberately not added a space.
+        abort if shutdown handler fails= True
+        abort if timeout handler fails = True
+        abort if stalled handler fails = True
+        abort if inactivity handler fails = False
+        mail to = eleanor.rigby@beatles.lv
+        mail from = fr.mckenzie@beatles.lv
+        mail footer = "Collecting The Rice"
+        mail smtp = 123.456.789.10
+        timeout = 30
+        inactivity = 30
+        abort on inactivity = 30
+    [[parameters]]
+    [[parameter templates]]
+    [[mail]]
+        task event mail interval    = PT4M # deliberately added lots of spaces.
+
+[scheduling]
+    max active cycle points = 5
+    hold after point = 20220101T0000Z
+    [[dependencies]]
+        [[[R1]]]
+            graph = foo
+
+
+
+[runtime]
+    [[MYFAM]]
+        extra log files = True
+        {% from 'cylc.flow' import LOG %}
+        script = {{HELLOWORLD}}
+        suite state polling = PT1H
```

Comment:
`suite state polling` was a section, not a setting

Reply:
🐆